src/snowflake/snowpark/_internal/server_connection.py

```diff
@@ -225,6 +227,10 @@ def upload_file(
                 raise ne.with_traceback(tb) from None
         else:
             uri = normalize_local_file(path)
+            if SNOWFLAKE_CONNECTOR_VERSION >= (3, 0, 3, None):
```

Comment:
Let's have a jira/todo here to remove this check, once connector is released?

Reply:
we can't remove this check after connector release right. We would also have to set minimum snowflake connector version to be >= 3.0.3. The current minimum required version is 2.7. What should be the policy to update the minimum required connector version? @sfc-gh-yixie @sfc-gh-mkeller 

https://snowflakecomputing.atlassian.net/browse/SNOW-769892

Reply:
I found a way using `inspect.signature` to see if a function accepts a certain keyword argument